src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -78,11 +81,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if ($input->getOption('no-warmup')) {
             $filesystem->rename($realCacheDir, $oldCacheDir);
         } else {
-            $warning = 'Calling cache:clear without the --no-warmup option is deprecated since version 3.3. Cache warmup should be done with the cache:warmup command instead.';
-
-            @trigger_error($warning, E_USER_DEPRECATED);
-
-            $io->warning($warning);
+            if ($this->warning) {
```

Comment:
This must be called after `$this->warmupCache($input, $output, $realCacheDir, $oldCacheDir);`, as the warning is populated inside `warmup`